packages/mempool/src/mempool.ts

```diff
@@ -134,16 +148,34 @@ export class MemoryMempool implements Mempool {
             ...this.store.dumpProcessing(),
             ...this.store.dumpSubmitted().map((sop) => sop.userOperation)
         ]
-        if (allOps.find((uo) => uo.userOperation.sender === op.sender && uo.userOperation.nonce === op.nonce)) {
-            return false
+        const oldUserOp = allOps.find(
+            (uo) => uo.userOperation.sender === op.sender && uo.userOperation.nonce === op.nonce
+        )
+        if (oldUserOp) {
+            const oldMaxPriorityFeePerGas = oldUserOp.userOperation.maxPriorityFeePerGas
+            const newMaxPriorityFeePerGas = op.maxPriorityFeePerGas
+            const oldMaxFeePerGas = oldUserOp.userOperation.maxFeePerGas
+            const newMaxFeePerGas = op.maxFeePerGas
+
+            const incrementMaxPriorityFeePerGas = (oldMaxPriorityFeePerGas * BigInt(10)) / BigInt(100)
+            const incrementMaxFeePerGas = (oldMaxFeePerGas * BigInt(10)) / BigInt(100)
+
+            if (
+                newMaxPriorityFeePerGas < oldMaxPriorityFeePerGas + incrementMaxPriorityFeePerGas ||
+                newMaxFeePerGas < oldMaxFeePerGas + incrementMaxFeePerGas
+            ) {
+                return false
+            }
+
+            this.store.removeOutstanding(oldUserOp.userOperationHash)
         }
 
         const hash = getUserOperationHash(op, this.entryPointAddress, this.publicClient.chain.id)
 
         this.store.addOutstanding({
             userOperation: op,
             userOperationHash: hash,
-            firstSubmitted: Date.now(),
+            firstSubmitted: oldUserOp ? oldUserOp.firstSubmitted : Date.now(),
             lastReplaced: Date.now()
         })
         this.monitor.setUserOperationStatus(hash, { status: "not_submitted", transactionHash: null })
```

Comment:
but the user operation is still not submitted, to a developer if they get a user operation hash back it means that the new user operation has been accepted and replaced the older one. But when they do a `pimlico_getUserOperationStatus` they should be informed that the replaced user op has still not been submitted and is with us right? What do you think?

Because replaced status won't convey user operation's submission status.